Import the data from the csv file into a dataframe and take a first look at it.

In [7]:
import pandas as pd

sleep_health_and_lifestyle_df = pd.read_csv("Sleep_health_and_lifestyle_dataset.csv")

sleep_health_and_lifestyle_df

Person ID  Gender  Age            Occupation  Sleep Duration  \
0            1    Male   27     Software Engineer             6.1   
1            2    Male   28                Doctor             6.2   
2            3    Male   28                Doctor             6.2   
3            4    Male   28  Sales Representative             5.9   
4            5    Male   28  Sales Representative             5.9   
..         ...     ...  ...                   ...             ...   
369        370  Female   59                 Nurse             8.1   
370        371  Female   59                 Nurse             8.0   
371        372  Female   59                 Nurse             8.1   
372        373  Female   59                 Nurse             8.1   
373        374  Female   59                 Nurse             8.1   

     Quality of Sleep  Physical Activity Level  Stress Level BMI Category  \
0                   6                       42             6   Overweight   
1                   6                       60             8       Normal   
2                   6                       60             8       Normal   
3                   4                       30             8        Obese   
4                   4                       30             8        Obese   
..                ...                      ...           ...          ...   
369                 9                       75             3   Overweight   
370                 9                       75             3   Overweight   
371                 9                       75             3   Overweight   
372                 9                       75             3   Overweight   
373                 9                       75             3   Overweight   

    Blood Pressure  Heart Rate  Daily Steps Sleep Disorder  
0           126/83          77         4200            NaN  
1           125/80          75        10000            NaN  
2           125/80          75        10000            NaN  
3           140/90          85         3000    Sleep Apnea  
4           140/90          85         3000    Sleep Apnea  
..             ...         ...          ...            ...  
369         140/95          68         7000    Sleep Apnea  
370         140/95          68         7000    Sleep Apnea  
371         140/95          68         7000    Sleep Apnea  
372         140/95          68         7000    Sleep Apnea  
373         140/95          68         7000    Sleep Apnea  

[374 rows x 13 columns]

We see different features such as gender, age, occupation, and so on. These feautures can be split into different groupings. To start, let us extract all the information of the participants and understand better who they are.

In [8]:
num_people = len(sleep_health_and_lifestyle_df)

min_age = sleep_health_and_lifestyle_df['Age'].min()
max_age = sleep_health_and_lifestyle_df['Age'].max()

occupations = sleep_health_and_lifestyle_df['Occupation'].unique()

# Print the results
print(f"Number of participants: {num_people}")
print(f"Range of ages: {min_age} - {max_age}")
print("Different occupations:", ', '.join(occupations))

Number of participants: 374
Range of ages: 27 - 59
Different occupations: Software Engineer, Doctor, Sales Representative, Teacher, Nurse, Engineer, Accountant, Scientist, Lawyer, Salesperson, Manager


With this information there are a couple of generalizations we can make. We know that, for the most part, all of these participants work white-collars job. The age range of participants are people who are in or near to the 'middle' of their life. There are only 374 participants, so the sample size in comparison to rest of the world is quite small. There is also a lack of information in where there participants hail from. 